### PR TITLE
VS C# Code Level Metrics

### PR DESCRIPTION
**Pull request by CalvinAllen**
<sup> Created from VS Code using [CodeStream](https://codestream.com/?utm_source=cs&utm_medium=pr&utm_campaign=github*com)</sup>

### CODE REVIEW

**Comment by CalvinAllen** [comment]
@bcanzanella I think this is ready to go - I did do some cleanup / etc.

File: vs/publishManifest.json [hunk 1]
```diff
@@ -1,14 +0,0 @@
-{
```
Comment (bcanzanella) [inline vs/publishManifest.json:1]: i think we need this file... pretty sure it's needed for our CI/CD deploy to marketplace pipeline

Comment (CalvinAllen) [reply vs/publishManifest.json:1]: Let me double check this, I think there were two of them - the other in the publish folder maybe?

File: vs/src/CodeStream.VisualStudio.Shared/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+//      Major Version
+//      Minor Version
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyFileVersion("1.0.0.0")]
```
Comment (bcanzanella) [inline vs/src/CodeStream.VisualStudio.Shared/Properties/AssemblyInfo.cs:35]: update these to the current version (13.0.1.0)

File: vs/src/CodeStream.VisualStudio.CodeLens/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+//      Minor Version
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+// TODO fix these
+[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyFileVersion("1.0.0.0")]
```
Comment (bcanzanella) [inline vs/src/CodeStream.VisualStudio.CodeLens/Properties/AssemblyInfo.cs:36]: update these to the current version (13.0.1.0)

File: vs/src/CodeStream.VisualStudio/UI/ToolWindows/CodeLevelMetricsDetails.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+						var componentModel = (IComponentModel)Package.GetGlobalService(typeof(SComponentModel));
+						var eventAggregator = componentModel.GetService<IEventAggregator>();
+						IDisposable d = null;
+						d = eventAggregator.GetEvent<WebviewDidInitializeEvent>().Subscribe(s => {
+							try {
+								_ = _codeStreamService.ViewMethodLevelTelemetryNotificationAsync(dataContext.Repo, dataContext.FunctionName, dataContext.NewRelicEntityGuid, dataContext.MetricTimeSliceNameMapping);
+								d?.Dispose();
+							}
+							catch (Exception ex) {
+								Log.Error(ex, $"{nameof(AddCodemarkCommandBase)} event");
```
Comment (bcanzanella) [inline vs/src/CodeStream.VisualStudio/UI/ToolWindows/CodeLevelMetricsDetails.xaml.cs:67]: nit pick on the `nameof`

**Review by bcanzanella — CHANGES_REQUESTED** [review]
couple of small changes, but otherwise looks awesome!

File: vs/publishManifest.json (hunk 1 above)
Comment (CalvinAllen) [reply vs/publishManifest.json]: You're right - the *other* files are duplicates / wrong.

File: vs/src/CodeStream.VisualStudio/UI/ToolWindows/CodeLevelMetricsDetails.xaml.cs (hunk 4 above)
Comment (CalvinAllen) [reply vs/src/CodeStream.VisualStudio/UI/ToolWindows/CodeLevelMetricsDetails.xaml.cs]: Fixed - also found an issue with that traced back to the RequestCodeReviewCommand

**Comment by CalvinAllen** [comment]
@bcanzanella All that should be resolved now

**Review by bcanzanella — APPROVED** [review]
🚀 🚀 🚀

**Pull request merged**